x-pack/plugins/cases/common/api/cases/case.ts

```diff
@@ -52,19 +60,19 @@ const CaseBasicRt = rt.strict({
   /**
    * The description of the case
    */
-  description: rt.string,
+  description: limitedStringSchema('description', 1, MAX_DESCRIPTION_LENGTH),
```

Comment:
We should not put the limitation in the `CaseBasicRt`. The reason is that is being used by `CaseAttributesRt` which is used by the cases client and the services to decode. If an existing case has a description with more than 30K characters, when a user tries to fetch it the case service will throw an error. We do not want to do that. You can use the `CasePostRequestRt` in the `CasePatchRequestRt`.

Reply:
Added all `caseBasicRt` types separately in  `CasePatchRequestRt ` with validation. 
Because in `CasePostRequestRt ` only few types are partial match where as in `CasePatchRequestRt` has all types as partial.